modules/imgproc/test/ocl/test_gftt.cpp

```diff
@@ -100,22 +101,27 @@ OCL_TEST_P(GoodFeaturesToTrack, Accuracy)
 
         std::vector<Point2f> upts, pts;
 
-        OCL_OFF(cv::goodFeaturesToTrack(src_roi, points, maxCorners, qualityLevel, minDistance, noArray()));
+        OCL_OFF(cv::goodFeaturesToTrack(src_roi, points, maxCorners, qualityLevel, minDistance, noArray(), 3, 0, 0.04, values));
         ASSERT_FALSE(points.empty());
+        ASSERT_FALSE(values.empty());
         UMatToVector(points, pts);
 
-        OCL_ON(cv::goodFeaturesToTrack(usrc_roi, upoints, maxCorners, qualityLevel, minDistance));
+        OCL_ON(cv::goodFeaturesToTrack(usrc_roi, upoints, maxCorners, qualityLevel, minDistance, noArray(), 3, 0, 0.04, uvalues));
         ASSERT_FALSE(upoints.empty());
+        ASSERT_FALSE(uvalues.empty());
         UMatToVector(upoints, upts);
 
+        ASSERT_EQ(pts.size(), values.size());
         ASSERT_EQ(upts.size(), pts.size());
+        ASSERT_EQ(uvalues.size(), values.size());
 
         int mistmatch = 0;
         for (size_t i = 0; i < pts.size(); ++i)
         {
-            Point2i a = upts[i], b = pts[i];
+            Point2f a = upts[i], b = pts[i];
 
-            bool eq = std::abs(a.x - b.x) < 1 && std::abs(a.y - b.y) < 1;
+            bool eq = std::abs(a.x - b.x) < 1 && std::abs(a.y - b.y) < 1 &&
+                    std::abs(values[i] - uvalues[i]) <= 3.f * FLT_EPSILON * std::max(values[i], uvalues[i]);
```

Comment:
New condition increases amount of mismatches. I propose to tune Eps a little to pass original check.

Reply:
OK, tests passes with 3.f * FLT_EPSILON relative error threshold.